conda_rpms/install.py

```diff
@@ -399,6 +403,129 @@ def symlink_conda(prefix, root_dir):
+    elif source_short_path.startswith('python-scripts/'):
+        bin_dir = 'bin'
+        return source_short_path.replace('python-scripts', bin_dir, 1)
+    else:
+        return source_short_path
+
+
+def compile_pyc(python_exe_full_path, py_full_path):
+    """
+    Modified version of conda/gateways/disk/create.py:compile_pyc at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe
+        * removed check that the pyc file already exists prior to compilation
+        * remove check that the pyc file has been created after the compilation
+        * changed log.trace -> log.info (log.trace not supported)
+
+    """
+    command = [python_exe_full_path, '-Wi', '-m', 'py_compile', py_full_path]
+    log.info(command)
+    subprocess.call(command)
+
+
+def parse_entry_point_def(ep_definition):
+    """
+    Copy of conda/core.path.py:parse_entry_point_def at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe
+
+    """
+    cmd_mod, func = ep_definition.rsplit(':', 1)
+    command, module = cmd_mod.rsplit("=", 1)
+    command, module, func = command.strip(), module.strip(), func.strip()
+    return command, module, func
+
+
+def make_executable(path):
+    """
+    Modified version of conda/gateways/disk/permissions.py:make_executable at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe
+        * changed lstat -> os.lstat
+        * changed chmod -> os.chmod
+        * changed log.trace -> log.info (log.trace not supported)
+
+    """
+    if isfile(path):
+        mode = os.lstat(path).st_mode
+        log.info('chmod +x %s', path)
+        os.chmod(path, S_IMODE(mode) | S_IXUSR | S_IXGRP | S_IXOTH)
+    else:
+        log.error("Cannot make path '%s' executable", path)
+
+
+def create_python_entry_point(target_full_path, python_full_path, module, func):
+    """
+    Modified version of conda/gateways/disk/create.py:create_python_entry_point
+    at be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe
+        * Removed check that the entry point already exists
+        * Removed the call that modifies a shebang>127 characters
+
+    """
+    import_name = func.split('.')[0]
+    pyscript = python_entry_point_template % {
+        'module': module,
+        'func': func,
+        'import_name': import_name,
+    }
+
+    if python_full_path is not None:
+        shebang = '#!%s\n' % python_full_path
+        # TODO : replace_long_shebang????
+    else:
+        shebang = None
+
+    with open(target_full_path, str('w')) as fo:
+        if shebang is not None:
+            fo.write(shebang)
+        fo.write(pyscript)
+
+    if shebang is not None:
+        make_executable(target_full_path)
+
+    return target_full_path
+
+def get_python_version(prefix):
+    """
+    Get the version of the python that is already linked in the environment.
+
+    """
+    py_ver = None
+    for dist in linked(prefix):
+        match = re.search('python-(\d+.\d+).\d+-\d+', dist)
+        if match:
+            py_ver = match.group(1)
```

Comment:
Personally, I'd put an else on this for loop to catch an issue with python not being installed.

Reply:
Alright
I have encountered the question of what to do with errors frequently in this.

I don't want an exception to be raised as then that would break the RPM installation (I think). I would prefer it to just fail silently, but then for us to be able to see why it failed by looking at logs (not that we can see them).

Reply:
>  I'd put an else on this for loop to catch an issue with python not being installed
How would you handle Python not being installed.
Is it enough to add something to the log that say "Python not already linked in the env"?